pkg/kubectl/cmd/apply_test.go

```diff
@@ -220,7 +219,7 @@ func TestApplyNonExistObject(t *testing.T) {
 		Client: fake.HTTPClientFunc(func(req *http.Request) (*http.Response, error) {
 			switch p, m := req.URL.Path, req.Method; {
 			case p == pathNameRC && m == "GET":
-				return &http.Response{StatusCode: 404}, errors.NewNotFound("ReplicationController", "")
+				return &http.Response{StatusCode: 404, Body: ioutil.NopCloser(bytes.NewReader(nil))}, nil
```

Comment:
Because this now implements a [`http.RoundTripper`](https://golang.org/pkg/net/http/#RoundTripper) interface. The documentation states that you can't return both a response and an error. This was causing a test failure.

Also, the documentation for [`http.Response`](https://golang.org/pkg/net/http/#Response) guarantees that the body is non-nil. This was causing a panic because our client relies on this being non-nil.